tsdb/querier.go

```diff
@@ -419,54 +412,64 @@ func inversePostingsForMatcher(ctx context.Context, ix IndexReader, m *labels.Ma
 		return ix.Postings(ctx, m.Name, m.Value)
 	}
 
-	vals, err := ix.LabelValues(ctx, m.Name)
-	if err != nil {
-		return nil, err
-	}
-
-	var res []string
+	var filter func(_ string) bool
 	// If the inverse match is ="", we just want all the values.
 	if m.Type == labels.MatchEqual && m.Value == "" {
-		res = vals
+		filter = func(_ string) bool { return true }
```

Comment:
I wonder if defining this as a global function would improve the performance:

```go
func all(_ string) bool { return true } 
//...
     filter = all
```

Reply:
It's possible, but none of my benchmark cases show a problem here.  When reading labels from a block's index the slow part is decoding varints.